src/renderer/containers/Bank/AddBankSigningKeyModal/index.tsx

```diff
@@ -0,0 +1,84 @@
+import React, {FC, useMemo} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+
+import {FormTextArea} from '@renderer/components/FormComponents';
+import Modal from '@renderer/components/Modal';
+import {SIGNING_KEY_LENGTH_ERROR, SIGNING_KEY_REQUIRED_ERROR} from '@renderer/constants/form-validation';
+import {useAddress} from '@renderer/hooks';
+import {getBankConfigs, getManagedBanks} from '@renderer/selectors';
+import {setManagedBank} from '@renderer/store/app';
+import {AppDispatch} from '@renderer/types';
+import {getKeyPairFromSigningKeyHex} from '@renderer/utils/signing';
+import yup from '@renderer/utils/yup';
+
+interface ComponentProps {
+  close(): void;
+}
+
+const initialValues = {
+  signingKey: '',
+};
+
+type FormValues = typeof initialValues;
+
+const AddBankSigningKeyModal: FC<ComponentProps> = ({close}) => {
+  const address = useAddress();
+  const dispatch = useDispatch<AppDispatch>();
+  const bankConfigs = useSelector(getBankConfigs);
+  const {
+    data: {node_identifier: nodeIdentifier},
+  } = bankConfigs[address];
+  const managedBanks = useSelector(getManagedBanks);
+  const managedBank = managedBanks[address];
+
+  const handleSubmit = ({signingKey}: FormValues): void => {
+    dispatch(
+      setManagedBank({
+        ...managedBank,
+        signing_key: signingKey,
+      }),
+    );
+    close();
+  };
+
+  const managedBankSigningKeys = useMemo(
+    () =>
+      Object.values(managedBanks)
+        .filter(({signing_key}) => !!signing_key)
```

Comment:
will we ever have a `managedBank` that doesn't have a `signing_key`? Should we change the type file to show that it is optional?

Reply:
When they don't have a `signing_key` we can just use an empty string so we only have to deal with 1 data type. Most banks won't have their `signing_key` filled out because users will probably treat managed banks more like bookmarks.